pkg/apis/performanceprofile/v2/performanceprofile_validation_test.go

```diff
@@ -369,6 +369,61 @@ var _ = Describe("PerformanceProfile", func() {
 			})
 		})
 	})
+
+	Describe("validation of validateFields function", func() {
+		It("should check all fields", func() {
+			reservedCPUs := CPUSet("")
+			isolatedCPUs := CPUSet("0-6")
+			offlinedCPUs := CPUSet("7")
+			profile.Spec.CPU.Reserved = &reservedCPUs
+			profile.Spec.CPU.Isolated = &isolatedCPUs
+			profile.Spec.CPU.Offlined = &offlinedCPUs
+
+			profile.Spec.MachineConfigLabel["foo"] = "bar"
+
+			incorrectDefaultSize := HugePageSize("!#@")
+			profile.Spec.HugePages.DefaultHugePagesSize = &incorrectDefaultSize
+
+			profile.Spec.WorkloadHints = &WorkloadHints{
+				RealTime: pointer.BoolPtr(false),
+			}
+			profile.Spec.RealTimeKernel = &RealTimeKernel{
+				Enabled: pointer.BoolPtr(true),
+			}
+
+			invalidVendor := "123"
+			invalidDevice := "0x12345"
+			profile.Spec.Net.Devices[0].InterfaceName = pointer.StringPtr("")
+			profile.Spec.Net.Devices[0].VendorID = pointer.StringPtr(invalidVendor)
+			profile.Spec.Net.Devices[0].DeviceID = pointer.StringPtr(invalidDevice)
+
+			errors := profile.validateFields()
+
+			// check validateCPUs errors
+			Expect(errors[0].Error()).To(ContainSubstring("reserved CPUs can not be empty"))
+
+			// check validateSelectors errors
+			Expect(errors).NotTo(BeEmpty(), "should have validation error when the profile has two machine config selectors")
+			Expect(errors[1].Error()).To(ContainSubstring("you should provide only 1 MachineConfigLabel"))
+
+			// check validateHuhePages erros
+			Expect(errors).NotTo(BeEmpty(), "should have validation error when default huge pages size has invalid value")
+			Expect(errors[2].Error()).To(ContainSubstring("hugepages default size should be equal"))
+
+			//TODO add validateNuma Tests
+
+			//check validateNet errors
+			Expect(errors[3].Error()).To(ContainSubstring(fmt.Sprintf("device name cannot be empty")))
+			Expect(errors[4].Error()).To(ContainSubstring(fmt.Sprintf("device vendor ID %s has an invalid format. Vendor ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidVendor)))
+			Expect(errors[5].Error()).To(ContainSubstring(fmt.Sprintf("device model ID %s has an invalid format. Model ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidDevice)))
+
+			// check validateWorkloadHints erros
+			Expect(errors).NotTo(BeEmpty())
+			Expect(errors[6].Error()).To(ContainSubstring("realtime kernel is enabled, but realtime workload hint is explicitly disable"))
+
+			Expect(len(errors)).To(Equal(7))
```

Comment:
this is fragile, and so is expecting the error to happen on a specific order, unless the number of errors and their ordering is part of the API contract (and it should not, because it leaks unnecessary implementation details).
It is better to just look for the errors we expect in the error list, regardless of the order and of the total count.
Arguably, also checking for the error string is fragile, but this is yet another story.

Reply:
You are right, I'll add an issue to track it.

Reply:
Mario, you are adding validation of workload hints, but the test here is actually not checking that part.

Reply:
(You have the other PR where some combination of workload hints is prohibited)

Reply:
Ok, NVM I totally missed the line 422.

Reply:
:D